### PR TITLE
Add reset button to flex options

### PR DESCRIPTION
**Pull request by tomusborne**
This adds a reset button to our flex options if one of them has a value.

This allows for a one-click reset instead of having to potentially clear three separate inputs.

It adds a `gblocks-utility-label` element we can reuse later if we want. Might be worth looking at merging something like this with the `UnitPicker` component.

### CODE REVIEW

File: src/blocks/container/edit.js [hunk 1]
```diff
@@ -636,9 +637,32 @@ class GenerateBlockContainer extends Component {
...
 									>
+										<div className="gblocks-utility-label">
+											<label
+												htmlFor="gblocks-flex-grow-desktop"
+												className="components-base-control__label"
+											>
+												{ __( 'Flex', 'generateblocks' ) }
+											</label>
+
+											{ ( hasValue( flexGrow ) || hasValue( flexShrink ) || hasValue( flexBasis ) ) &&
```
Comment (JeanPaiva) [inline src/blocks/container/edit.js:649]: I don't think it's necessary to check for values to display the icon, the button can always be displayed in my opinion.

File: src/blocks/container/edit.js [hunk 2]
```diff
@@ -636,9 +637,32 @@ class GenerateBlockContainer extends Component {
...
+												<Tooltip text={ __( 'Reset', 'generateblocks' ) } position="top">
+													<Button
+														className="gblocks-inline-reset-button"
+														icon={ getIcon( 'refresh' ) }
```
Comment (JeanPaiva) [inline src/blocks/container/edit.js:653]: I don't think the icon matches exactly the intent it seems you can go back to old values. Maybe changing to a `delete` (backspace) icon of Feather icons. This is totally opinionated though.

File: src/blocks/container/edit.js [hunk 3]
```diff
@@ -825,9 +849,32 @@ class GenerateBlockContainer extends Component {
 
 									<BaseControl
 										className="gblocks-flex-controls"
-										label={ __( 'Flex', 'generateblocks' ) }
-										id="gblocks-flex-grow-tablet"
 									>
+										<div className="gblocks-utility-label">
```
Comment (JeanPaiva) [inline src/blocks/container/edit.js:853]: This hole block should be moved to a component, later on, to re-use logic and avoid code duplication.

**Review by JeanPaiva — COMMENTED** [review]
Testing is ok tho.

File: src/blocks/container/edit.js (hunk 1 above)
Comment (tomusborne) [reply src/blocks/container/edit.js:649]: Agreed, was getting a little fancy haha

File: src/blocks/container/edit.js (hunk 2 above)
Comment (tomusborne) [reply src/blocks/container/edit.js:653]: What are other plugins doing? Wonder if Gutenberg itself is using an "undo" or "clear" icon.

File: src/blocks/container/edit.js (hunk 3 above)
Comment (tomusborne) [reply src/blocks/container/edit.js:853]: Agreed - it should be merged with the `UnitPicker` component.

File: src/blocks/container/edit.js (hunk 2 above)
Comment (JeanPaiva) [reply src/blocks/container/edit.js:658]: I've no idea, but we can leave it like this for now, if users complaining about having confusion we can change it.

Comment (tomusborne) [reply src/blocks/container/edit.js]: Better or worse now?

Comment (JeanPaiva) [reply src/blocks/container/edit.js]: Yes, it's better IMO. 👍

**Review by JeanPaiva — APPROVED** [review]
All good.

**Pull request merged**